### PR TITLE
Yup, Still More Script Commands

### PR DESCRIPTION
**Pull request by ravepossum**
Documents and renames the following script commands:
- `ScrCmd_0D3` -> `BufferTMHMMoveName`
- `ScrCmd_0D4` -> `BufferMoveName`
- `ScrCmd_0DF` -> `BufferUndergroundGoodsName`
- `ScrCmd_0E0` -> `BufferUndergroundTrapName`
- `ScrCmd_0E1` -> `BufferUndergroundItemName`
- `ScrCmd_0EC` -> `CheckWonBattle`
- `ScrCmd_0ED` -> `CheckLostBattle`
- `ScrCmd_2BC` -> `CheckDidNotCapture`
- `ScrCmd_119` -> `CheckPartyPokerus`
- `ScrCmd_133` -> `RegisterPoketchApp`
- `ScrCmd_14D` -> `GetPlayerGender`
- `ScrCmd_14E` -> `HealParty`
- `ScrCmd_15B` -> `CheckBadge`
- `ScrCmd_177` -> `GetPartyCount`
- `ScrCmd_19A` -> `GetPartyCountHatched`
- `ScrCmd_19B` -> `CountAliveMonsExcept`
- `ScrCmd_1AD` -> `Dummy1AD`
- `ScrCmd_1E5` -> `IncrementGameRecord`
- `ScrCmd_1F7` -> `SurvivePoison`
- `Dummy1` -> `Dummy1F9` (renaming per new dummy function naming scheme)

Also adds a new constgen enum for badges and renames the `playerWonBattle` script manager member to the more appropriate name `battleResult`.

### CODE REVIEW

File: consts/scrcmd.json [hunk 1]
```diff
@@ -8,6 +8,19 @@
...
+            "type": "enum",
+            "values": [
+                "BADGE_ID_COAL",
+                "BADGE_ID_FOREST",
+                "BADGE_ID_COBBLE",
+                "BADGE_ID_FEN",
+                "BADGE_ID_RELIC",
+                "BADGE_ID_MINE",
+                "BADGE_ID_ICICLE",
+                "BADGE_ID_BEACON"
```
Comment (lhearachel) [inline consts/scrcmd.json:22]: **quibble**: I wonder if it's reasonable to provide aliases for these. I have a hard time remembering which is which. 😅 

**chore**: Add a `MAX_BADGES` or somesuch to this enum.

File: src/unk_0204E3CC.c [hunk 2]
```diff
@@ -60,15 +60,15 @@ BOOL ScrCmd_15A(ScriptContext *param0)
...
+BOOL ScrCmd_CheckBadge(ScriptContext *ctx)
 {
-    u16 v0 = ScriptContext_GetVar(param0);
-    u16 *v1 = ScriptContext_GetVarPointer(param0);
+    u16 badgeNum = ScriptContext_GetVar(ctx);
+    u16 *destVar = ScriptContext_GetVarPointer(ctx);
 
-    GF_ASSERT(v0 < 8);
-    *v1 = TrainerInfo_HasBadge(SaveData_GetTrainerInfo(param0->fieldSystem->saveData), v0);
+    GF_ASSERT(badgeNum < 8);
```
Comment (lhearachel) [inline src/unk_0204E3CC.c:68]: **chore**: Use `MAX_BADGES` constant here

**Review by lhearachel — APPROVED** [review]
Generally looks good to me. Just a few minor things.

File: consts/scrcmd.json (hunk 1 above)
Comment (ravepossum) [reply consts/scrcmd.json:22]: Would you mind giving me an example of what you mean by alias here?

**Comment by ravepossum** [comment]
Both comments should be addressed now, lmk if you had something different in mind.

**Pull request merged**